core/common/src/main/java/alluxio/grpc/GrpcManagedChannelPool.java

```diff
@@ -408,6 +415,8 @@ public int hashCode() {
           .append(mMaxInboundMessageSize)
           .append(mFlowControlWindow)
           .append(mPoolKey)
+          .append(mHealthCheckTimeoutMs)
```

Comment:
My only concern is that if we are to modify the timeouts that they wouldn't be respected within the same JVM. i.e. if one client chooses a timeout of 10min and the other with a timeout of 30s and the client with 30s timeout shuts down its channel and it takes 10 minutes- then what's the point of being able to configure the value?

Either way I understand the concern. From a performance standpoint I could see how creating unnecessary channels could be an issue. From a functionality standpoint though I think that the timeout should be included otherwise users' timeouts may not be respected which goes against the principle of instanced configuration.

My opinion is that the functionality here is more important than the performance. I would say it's unlikely a user is going to modify the timeout to include two different values within the same JVM. Keeping the code there ensures that those timeouts are respected. I think if a user does choose to do so, they are responsible for any performance decreases.

Reply:
I see. because the values are on the `channelKey`, and we're not retrieving the old `channelKey` whenever we shut down the channel it will use whatever timeout the last channel user configured it for. That's good.

It looks like the health check timeout will work the same way because the health check is only used when acquiring the managed which comes from a new channel key.

Reply:
It's more complicated with health check. Health check will be performed on existing managedchannels as well. 
However, since health check is not something that the client is controlling, I think we should be fine treating it as suggestion.

Reply:
If the health check is something that's being done on the server side - should there be a separate key for server vs client configuration? Right now on the client side I only see the value being used by `waitForChannelReady` which is only used in `acquireManagedChannel`

Reply:
I'm not sure. Health check can be done in all alluxio processes. I'd vote for a unified default value.